engine/lib/generic_printer/generic_printer.ml

```diff
@@ -594,6 +600,37 @@ module Make (F : Features.T) = struct
             self#concrete_ident ~local id)
           ast_position id
 
+      method _do_not_override_lazy_of_global_ident ast_position
+          (id : global_ident) : global_ident lazy_doc =
+        lazy_doc
+          (fun (id : global_ident) ->
+            match id with
+            | `Concrete cid ->
+                (self#_do_not_override_lazy_of_concrete_ident ast_position cid)
+                  #p
+            | `Primitive _prim_id -> string "(*TODO*) prim_id"
+            | `TupleType 0 -> string "unit"
+            (* | `TupleCons n when n <= 1 -> *)
+            (*   Error.assertion_failure span *)
+            (*     ("Got a [TupleCons " ^ string_of_int n ^ "]") *)
+            (* | `TupleType n when n <= 14 -> *)
+            (*   F.lid [ "FStar"; "Pervasives"; "tuple" ^ string_of_int n ] *)
+            (* | `TupleCons n when n <= 14 -> *)
+            (*   F.lid [ "FStar"; "Pervasives"; "Mktuple" ^ string_of_int n ] *)
+            (* | `TupleType n | `TupleCons n -> *)
+            (*   let reason = "F* doesn't support tuple of size greater than 14" in *)
+            (*   Error.raise *)
+            (*     { *)
+            (*       kind = UnsupportedTupleSize { tuple_size = Int64.of_int n; reason }; *)
+            (*       span; *)
+            (*     } *)
+            (* | `TupleField _ | `Projector _ -> *)
+            (*   Error.assertion_failure span *)
+            (*     ("pglobal_ident: expected to be handled somewhere else: " *)
+            (*      ^ show_global_ident id) *)
+            | _ -> string "(* TODO *) error global ident type?")
+          ast_position id
```

Comment:
`_do_not_override_lazy_of_global_ident` was undefined?

Reply:
I could not find a definition, but it seems most places it was only `Concrete` that was used, and then call directly to `_do_not_override_lazy_of_concrete_ident`

Reply:
I think my issue might have been that `field` was evaluated, now that it is lazy, the errors I had, no longer happens, since I never call `#p`.